.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,82 @@
+// !/bin/bash
+//  SPDX-FileCopyrightText: Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//  SPDX-License-Identifier: Apache-2.0
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+{
+    "name": "Morpheus",
+    "build": {
+        "dockerfile": "Dockerfile"
+    },
+
+    "hostRequirements": {
+        "gpu": true
+    },
+
+    "capAdd":[
+        "SYS_NICE",
+        "SYS_PTRACE"
+    ],
+
+    "securityOpt": [
+        "seccomp=unconfined"
+    ],
+
+    "runArgs": [
+        "--network=morpheus"
+    ],
+
+    "containerEnv": {
+        "HOST_MORPHEUS_ROOT": "${localWorkspaceFolder}",
+        "MORPHEUS_ROOT": "/workspaces/morpheus"
+    },
+
+    "postStartCommand": "${MORPHEUS_ROOT}/.devcontainer/bin/setup-morpheus-env",
+
+    "initializeCommand": [ "./.devcontainer/init.sh" ],
+
+    "remoteUser": "coder",
+
+    "mounts": [
+        {"type":"bind", "source": "/var/run/docker.sock", "target": "/var/run/docker.sock"},
+        {"type":"bind", "source": "${localWorkspaceFolder}/.cache/conda", "target": "/home/coder/.conda"}
+    ],
+
+    "features": {
+        "ghcr.io/devcontainers/features/common-utils:1": {
+            "uid": "1000",
+            "gid": "1000",
+            "username": "coder"
+        },
+        "ghcr.io/devcontainers/features/docker-from-docker": {}
+    },
+
+    "customizations": {
+        "vscode": {
+            "extensions": [
+				"eamodio.gitlens",
+				"llvm-vs-code-extensions.vscode-clangd",
+				"ms-python.python",
+				"ms-vscode.cpptools",
+				"ms-vscode.cmake-tools",
+				"nvidia.nsight-vscode-edition",
+				"twxs.cmake",
+				"xaver.clang-format"
+			]
+        }
+    },
+    "settings": {
+        "C_Cpp.intelliSenseEngine": "disabled",
+        "python.terminal.activateEnvironment": false
+    }
```

Comment:
How do these settings interact with the settings in `.vscode/settings.json` and the settings in `morpheus.code-workspace`?

Reply:
From highest to lowest priority, all are considered:
```
/workspaces/morpheus/.vscode/settings.json (workspace dir) >
/workspaces/morpheus/morpheus.code-workspace (workspace file) >
/home/coder/.vscode-server/data/Machine/settings.json (devcontainer settings) >
/home/charris/.config/Code/User/settings.json (local user settings)
```

Your local user settings are considered first, then overridden by the devcontainer settings, which are overridden by the workspace file, which are then overridden by the `.vscode/settings.json` file.

So it's important that the devcontainer only contain settings that are critical to function. For instance, `python.terminal.activateEnvironment` here is important because otherwise it tries to use the wrong conda environment. `C_Cpp.intelliSenseEngine` is important because we include both the C++ and clangd plugins in the devcontainer, and clangd should take priority.